xclim/indices/generic.py

```diff
@@ -88,25 +88,26 @@ def select_resample_op(
     da = select_time(da, **indexer)
     r = da.resample(time=freq)
     if isinstance(op, str):
-        return getattr(r, op)(dim="time", keep_attrs=True)
-
-    return r.map(op)
+        out = getattr(r, op)(dim="time", keep_attrs=True)
+    else:
+        with xr.set_options(keep_attrs=True):
+            out = r.map(op)
+            op = op.__name__
+    return to_agg_units(out, da, op)
 
 
 def doymax(da: xr.DataArray) -> xr.DataArray:
     """Return the day of year of the maximum value."""
     i = da.argmax(dim="time")
     out = da.time.dt.dayofyear.isel(time=i, drop=True)
-    out.attrs.update(units="", is_dayofyear=np.int32(1), calendar=get_calendar(da))
-    return out
+    return to_agg_units(out, da, "doymax")
```

Comment:
using `to_agg_units` directly in `doymax`. Does that make things to obscure? I thought it might be better like this to avoid having the same info repeating in two places

Reply:
Hum, looks kinda obscure to me as `doymax` is to be applied repeatedly on each resampling group. Seems strange to me that it would set the attributes itself ? (I see that this was the case before your change)

Reply:
The units are always obvious following the doymax operation no? Is it a question of organizing philosophy, where units should be treated? Maybe I don't understand the subtlety if "applying this repeatedly on each resampling group"

Reply:
I meant by that that `doymax` is not a full operation, it's one that gets applied in a `resample` operation. But now that I think of it, there could be use cases where this is not true, and then we would absolutely want the units to be applied in `doymax`. I retract my comment, your implementation makes sense.